### DISCUSSION TITLE
ユーザー名の読み上げ、GPUモードについて

### DISCUSSION THREAD

**Discussion by babbieeee** [ciffelia/koe#45] [Q&A] [answered]
はじめまして。質問させていただきます。

ciffelia様のkoeの説明を大変参考にさせていただき、discordのテキスト読み上げができるところまで構築することができました。
件名の通りですが、
・discordのテキスト読み上げの際、ユーザー名も読み上げてしまうのでこれを読み上げないようにしたい
・VOICEVOXのCPU、GPUモードの切り替えをするにはどうしたらよいか
上記の2点について解決できずにいます。
これらを変更するにはどうしたらよいでしょうか。ご教示いただけますと幸いです。

dockerやVOICEVOXについても完全な初心者ですので、初歩的な質問で申し訳ありません。

**Comment by ciffelia** [comment, marked as answer]
ユーザー名を読み上げないようにするには、以下の部分（`fn should_read_author_name`の中身）を`false`に書き換えてkoeのイメージをビルドすればよいはずです。
https://github.com/ciffelia/koe/blob/main/crates/koe/src/message/read.rs#L46-L52

ビルドは`docker build -t 任意のタグ .`で実行できます。ビルドしたイメージを使うには、`docker-compose.yml`の`ghcr.io/ciffelia/koe`をビルド時に指定したタグに書き換えてください。

GPUモードにするには、`docker-compose.yml`の`cpu`を`nvidia`に書き換えてください。

**Reply by babbieeee** [reply]
返信ありがとうございました。
無事、ご教示いただいた点を変更し、思うような動作へ変更することができました。
ありがとうございました！
